website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,478 @@
+                }},
+                    m('i.fa.fa-angle-down'));
+            }
+        }
+
+        function sortDateAsc(){
+            if (ctrl.loadingComplete()){
+                 return m('button', {'class': ctrl.colorSortButtons('dateAsc'), onclick: function() {
+                     ctrl.sortBySortState(ctrl.sortState('dateAsc'));
+                 }},
+                     m('i.fa.fa-angle-up'));
+            }
+        }
+
+        function sortDateDesc(){
+            if (ctrl.loadingComplete()){
+                return m('button', {'class': ctrl.colorSortButtons('dateDesc'), onclick: function() {
+                    ctrl.sortBySortState(ctrl.sortState('dateDesc'));
+               }},
+                    m('i.fa.fa-angle-down'));
+            }
+        }
+
+        // Sort button for xs screen
+        function ascending() {
+            if (ctrl.loadingComplete()){
+                return m('button', {'class': ctrl.colorSortButtonsXS('Asc'), onclick: function() {
+                     ctrl.directionSort('Asc');
+                     ctrl.sortDirectionGivenField();
+                }},
+                     m('i.fa.fa-angle-up'));
+            }
+        }
+
+        // Sort button for xs screen
+        function descending() {
+            if (ctrl.loadingComplete()){
+                return m('button', {'class': ctrl.colorSortButtonsXS('Desc'), onclick: function() {
+                    ctrl.directionSort('Desc');
+                    ctrl.sortDirectionGivenField();
+                }},
+                     m('i.fa.fa-angle-down'));
+            }
+        }
+
+        // Dropdown for XS screen - if sort on title on large screen, when resize to xs, 'title' is default selected
+        function defaultSelected() {
+            var selected = ctrl.preSelectField();
+            if (selected === 'alpha') {
+                return [m('option', {value: 'alpha', selected:'selected'}, 'Title'), m('option', {value: 'date'}, 'Modified')];
+            }
+            else {
+                return [m('option', {value: 'alpha'}, 'Title'), m('option', {value: 'date', selected:'selected'}, 'Modified')];
+            }
+        }
+
+        function searchBar() {
+            if (ctrl.loadingComplete()){
+                return m('div.m-v-sm.quick-search-input', [
+                    m('input[type=search]', {'id': 'searchQuery', 'class': 'form-control', placeholder: 'Quick search projects', onkeyup: function(search) {
+                        ctrl.filter(search.target.value);
+                        ctrl.quickSearch();
+                    }}),
+                    m('span', {onclick: function() {
+                        ctrl.filter('');
+                        document.getElementById('searchQuery').value = '';
+                        ctrl.quickSearch();
+                    }},  m('button', m('i.fa.fa-times')))
+                ]);
+            }
+        }
+
+        function displayNodes() {
```

Comment:
Rather than have these private functions for returning part of the view, it is better to separate these out into separate, stateless subcomponents.

For example:

```js
var NodeList = {
    view: function(ctrl, args) {
        if (args.eligibleNodes().length === 0 && args.filter() != null) {
            return m('.row.m-v-sm', m('.col-sm-10.col-sm-offset-1',
                m('.row',
                    m('.col-sm-12', [m('p.fa.fa-exclamation-triangle'), m('em', 'No results found!')])
                ))
            );
        }
        else {
            return args.eligibleNodes().slice(0, args.countDisplayed()).map(function(n){
                var project = args.nodes()[n];
                var numContributors = project.embeds.contributors.links.meta.total;
                var onClickNode = args.onClickNode.bind(project);
                return m('.m-v-sm.node-styling', {onclick: onClickNode}, m('.row',
                    [
                        // If you want, you could break out single node items into its own component, e.g.
                         // m.component(NodeListItem, {data: project})
                    ]
                ));
            });
        }

    }
}

// ..
m.component(NodeList, {
    nodes: ctrl.nodes(), 
    filter: ctrl.filter(),
    countDisplayed: ctrl.countDisplayed(),
    eligibleNodes: ctrl.eligibleNodes(),
    onClickNode: function(node) {
        // redirect to node
    }
})
```

Spend an hour refactoring in this way. If it takes too much time, feel free to defer, but be thinking about how you can break this code up into smaller components. 

This makes for more efficient (because you're not redeclaring these private functions), readable (separating concerns), testable (small components are easy to test), and reusable code.